R/importFilterHypermutators.R

```diff
@@ -0,0 +1,201 @@
+# importFilterHypermutators.R
+
+#' Filters out hypermutators
+#'
+#' \code{importFilterHypermutators} identifies the number of SNV and CNA
+#' mutations for each sample and removes the samples from rSNV and rCNA
+#' datasets. Default threshold for removal is 400 mutations per sample.
+#'
+#' @param rSNVFileIn A vector of local file names of rSNV
+#' @param rCNAFileIn A vector of local file names of rCNA
+#' @param dOut Directory to store output, defaults to getwd().
+#' @param xS Mutation threshold. 400 by default.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @param writeDetailedLog Flag for extra details about log. TRUE by default.
+#'
+#' @examples
+#' \dontrun{
+#'     importFilterHypermutators(rSNVFileIn, rCNAFileIn, dOut, xS)
+#' }
+#'
+#' @export
+importFilterHypermutators <- function(rSNVFileIn = c(),
+                                      rCNAFileIn = c(),
+                                      dOut = getwd(),
+                                      xS = 400,
+                                      silent = FALSE,
+                                      writeLog = TRUE,
+                                      writeDetailedLog = TRUE) {
+
+    ## VALIDATE PARAMS ########
+    # make sure that files actually exist
+    for (file in rSNVFileIn) {
+        r <- .checkArgs(file, like = "FILE_E", checkSize = TRUE)
+        if (length(r) > 0) {
+            stop(r)
+        }
+    }
+
+    for (file in rCNAFileIn) {
+        r <- .checkArgs(file, like = "FILE_E", checkSize = TRUE)
+        if (length(r) > 0) {
+            stop(r)
+        }
+    }
+
+    # make sure that output directory is valid
+    r <- .checkArgs(dOut, like = "DIR", checkSize = TRUE)
+    if(length(r) > 0) {
+        stop(r)
+    }
+
+    if (!is.double(xS)) {
+        stop(xS)
+    }
+
+    # set up hashes
+    hashTable <- new.env(hash = TRUE)
+    totalSamples <- 0
+
+    ## COUNT MUTATIONS ########
+    # for each file in rCNAFileIn
+    for (file in rCNAFileIn) {
+        # readRDS, count gene CNAs and add to hash for sample @ CNA and nMUT
+        rCNA <- readRDS(file)
+        for (sample in names(rCNA)) {
+            print(sample)
+            totalSamples <- totalSamples + 1
+            # if key is in hash, increment CNA and nMUT
+            if (!is.null(hashTable$sample)) {
```

Comment:
That looks kind of weird. Did you mean this as an if/else construct with line 78 in the else-block?